sentry-ruby/lib/sentry/faraday.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Sentry
+  module Faraday
+    class Instrumenter
+      # Sentry.with_child_span(op: OP_NAME, start_timestamp: Sentry.utc_now.to_f, origin: SPAN_ORIGIN) do |sentry_span|
+      #   request_info = extract_request_info(req)
+
+      #   if propagate_trace?(request_info[:url], Sentry.configuration)
+      #     set_propagation_headers(req)
+      #   end
+
+      #   super.tap do |res|
+      #     record_sentry_breadcrumb(request_info, res)
+
+      #     if sentry_span
+      #       sentry_span.set_description("#{request_info[:method]} #{request_info[:url]}")
+      #       sentry_span.set_data(Span::DataConventions::URL, request_info[:url])
+      #       sentry_span.set_data(Span::DataConventions::HTTP_METHOD, request_info[:method])
+      #       sentry_span.set_data(Span::DataConventions::HTTP_QUERY, request_info[:query]) if request_info[:query]
+      #       sentry_span.set_data(Span::DataConventions::HTTP_STATUS_CODE, res.code.to_i)
+      #     end
+      #   end
+      # end
+      #
+      #  #<struct Faraday::Env
+      #  method=:get,
+      #  request_body=nil,
+      #  url=#<URI::HTTP http://example.com>,
+      #  request=#<struct Faraday::RequestOptions params_encoder=nil, proxy=nil, bind=nil, timeout=nil, open_timeout=nil, read_timeout=nil, write_timeout=nil, boundary=nil, oauth=nil, context=nil, on_data=nil>,
+      #  request_headers={"User-Agent"=>"Faraday v2.10.0"},
+      #  ssl=#<struct Faraday::SSLOptions verify=nil, verify_hostname=nil, ca_file=nil, ca_path=nil, verify_mode=nil, cert_store=nil, client_cert=nil, client_key=nil, certificate=nil, private_key=nil, verify_depth=nil, version=nil, min_version=nil, max_version=nil>,
+      #  parallel_manager=nil,
+      #  params=nil,
+      #  response=nil,
+      #  response_headers=nil,
+      #  status=nil,
+      #  reason_phrase=nil,
+      #  response_body=nil>]
+      def self.instrument(name, env, &block)
+        Sentry.with_child_span(op: "http.faraday", start_timestamp: Sentry.utc_now.to_f, origin: "auto.http.faraday") do |sentry_span|
```

Comment:
we have some conventions here
https://develop.sentry.dev/sdk/performance/span-operations/#web-server
so this should be `http.client` or `http.client.faraday` if you want to be more precise.